lifecycle-operator/apis/lifecycle/v1beta1/common/common.go

```diff
@@ -126,6 +130,13 @@ func GetOverallState(s StatusSummary) KeptnState {
 	return StateSucceeded
 }
 
+func GetOverallStateBlockedDeployment(state KeptnState, blockedDeployment bool) KeptnState {
```

Comment:
since it is used wrapping the function before this one as first parameter, would it make sense to combine the two?
i.e., having only `GetOverallState(StatusSummary)` in which we change only the failed branch?

Reply:
It may be a bit confusing why this one accepts a `KeptnState` as first parameter, while the similarly named `GetOverallState` takes a `StatusSummary`. Can we change this function to take a `StatusSummary`, then internally call `GetOverallState` and add the additional logic around that?
This way the use of this function will also be simplified a little (e.g. https://github.com/keptn/lifecycle-toolkit/pull/3113/files#diff-81b0d3d25ca935b905c5955c530cc26e742fee2fff8e606f9ea89edeb653c2e8R32), where we could just use `apicommon.GetOverallStateBlockedDeployment(state, r.Config.GetBlockDeployment())`, instead of `apicommon.GetOverallStateBlockedDeployment(apicommon.GetOverallState(state), r.Config.GetBlockDeployment())` first

Reply:
sure, I can adapt this

Reply:
adapted